tekton/pipeline_run.go

```diff
@@ -93,6 +93,10 @@ func (i *InternalRequestPipelineRun) WithInternalRequest(internalRequest *v1alph
 		i.Spec.Timeouts = &internalRequest.Spec.Timeouts
 	}
 
+	if internalRequest.Spec.ServiceAccount != "" {
```

Comment:
You can assign it directly. Empty string is the default value so it won't get added.

Reply:
fixed